config/src/main/java/org/springframework/security/config/annotation/web/configurers/ExpressionUrlAuthorizationConfigurer.java

```diff
@@ -308,27 +320,30 @@ public AuthorizedUrl not() {
 
 		/**
 		 * Shortcut for specifying URLs require a particular role. If you do not want to
-		 * have "ROLE_" automatically inserted see {@link #hasAuthority(String)}.
+		 * have role prefix (default "ROLE_") automatically inserted see
+		 * {@link #hasAuthority(String)}.
 		 * @param role the role to require (i.e. USER, ADMIN, etc). Note, it should not
-		 * start with "ROLE_" as this is automatically inserted.
+		 * start with role prefix as this is automatically inserted.
 		 * @return the {@link ExpressionUrlAuthorizationConfigurer} for further
 		 * customization
 		 */
 		public ExpressionInterceptUrlRegistry hasRole(String role) {
-			return access(ExpressionUrlAuthorizationConfigurer.hasRole(role));
+			return access(ExpressionUrlAuthorizationConfigurer
+					.hasRole(ExpressionUrlAuthorizationConfigurer.this.REGISTRY.rolePrefix, role));
```

Comment:
Suggest extracting `ExpressionUrlAuthorizationConfigurer.this.REGISTRY.rolePrefix` to constant or method.

Reply:
It only appear 2 times, I'm not sure it should extract as
```java
private final String rolePrefix = ExpressionUrlAuthorizationConfigurer.this.REGISTRY.rolePrefix;
```
leave it to @jzheaux to decide.

Reply:
I think what I would do is move the `rolePrefix` to be a member of `ExpressionUrlAuthorizationConfigurer` and the `ApplicationContext` support to `ExpressionUrlAuthorizationConfigurer`'s constructor.

The registry doesn't seem to have any need on its own for a role prefix, so it's best to leave it out.

Then, I'd change the references to `ExpressionUrlAuthorizationConfigurer.this.rolePrefix` instead of `ExpressionUrlAuthorizationConfigurer.this.REGISTRY.rolePrefix`.

Reply:
@jzheaux Updated.